doc/rst/technotes/gpu.rst

```diff
@@ -261,34 +220,77 @@ For more examples see the tests under |multi_locale_dir|_ available from our `pu
-unified-memory based approach.
-
-To use this new strategy set the environment variable ``CHPL_GPU_MEM_STRATEGY``
-to ``array_on_device``.  For more examples that work with this strategy see the
-tests under |page_lock_mem_dir|_  available from our `public Github repository
-<https://github.com/chapel-lang/chapel>`_.
-
-.. |page_lock_mem_dir| replace:: ``test/gpu/native/page-locked-mem/``
-.. _page_lock_mem_dir: https://github.com/chapel-lang/chapel/tree/main/test/gpu/native/page-locked-mem
+The ``CHPL_GPU_MEM_STRATEGY`` environment variable can be used to choose between
+two different memory strategies.
+
+The current default strategy is ``unified_memory``. This strategy stores data
+that is allocated on a GPU sublocale (i.e. ``here.gpus[0]``).  Under unified
+memory the underlying GPU implementation implicitly manages the migration of
+data to and from the GPU as necessary.
+
+The alternative is to set the environment variable explicitly to
+``array_on_device``. This strategy stores array data directly on the device and
+store other data on the host.  There are multiple benefits to using this
+strategy including that it enables users to have more explicit control over
+memory management, may be required for Chapel to interoperate with various
+third-party communication libraries, and may be necessary to achieve good
+performance. As such it may become the default memory strategy we use in the
+future. Be aware though that because this strategy is relatively new addition it
+hasn't been as thoroughly tested as our unified memory based approach.
 
 Note that host data can be accessed from within a GPU eligible loop running on
 the device via a direct-memory transfer.
 
-One limitation with memory access in this mode is that we do not support direct
-reads or writes from the host into individual elements of array data allocated
-on the GPU (e.g.  ``use(A[i])`` or ``A[i] = ...``). Array data accessed "as a
-whole" (e.g. ``writeln(A)``) will work, however.
+Debugger and Profiler Support for NVIDIA
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As of Chapel 1.30.0 ``cuda-gdb`` and `NVIDIA NSight Compute
+<https://developer.nvidia.com/nsight-compute>`_ can be used to debug and profile
+GPU kernels. We have limited experience with both of these tools.  However,
+compiling with ``-g`` and running the application in ``cuda-gdb`` help uncover
+segmentation faults coming from GPU kernels.
+
+Similarly, NSight Compute can be used to collect detailed performance metrics
+from GPU kernels generated by the Chapel compiler. By default, using ``-g`` only
+enables Chapel line numbers to be associated with performance metrics, however
+it thwarts optimizations done by the backend assembler. In our experience, this
+can reduce execution performance significantly, making profiling less valuable.
+To avoid this, please use ``--gpu-ptxas-enforce-optimization`` while compiling
+alongside ``-g``, and of course, ``--fast``.
+
+Known Limitations
+-----------------
```

Comment:
I'm questioning whether it would also be worthwhile to link to our public tracker for GPU related bugs:

https://github.com/chapel-lang/chapel/issues?q=is%3Aopen+label%3A%22area%3A+GPU+Support%22+label%3A%22type%3A+Bug%22

Reply:
That's good idea. How about I just point to the "GPU Support" label in general (no "Bug" label) with a more general statement like "Use this label to search for issues related to GPU support"? Technically it is not "Known Limitations", though. Maybe I do that at the very end as a standalone statement?

Reply:
I'll probably create "Further Information" header and start a list there (maybe I'll move the reference to release notes there, too.